sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SymbolicExecution/ExplodedGraphTests.cs

```diff
@@ -935,5 +935,93 @@ public void Test(string s)
+                    }
+                    if (args.Instruction.ToString() == "k = default")
+                    {
+                        args.ProgramState.GetSymbolValue(kSymbol).Should().NotBeNull();
+                        kSymbol.HasConstraint(ObjectConstraint.NotNull, args.ProgramState).Should().BeFalse();
+                        kSymbol.HasConstraint(ObjectConstraint.Null, args.ProgramState).Should().BeTrue();
+                    }
+                };
+
+            explodedGraph.Walk();
+            numberOfProcessedInstructions.Should().Be(6);
+        }
+
+        [TestMethod]
+        [TestCategory("Symbolic execution")]
+        public void ExplodedGraph_TupleExpressions()
+        {
+            var testInput = "var myTuple = (1, 2); (object a, object b) c = (1, null); (object d, object e) = (1, null);";
+            var method = ControlFlowGraphTest.CompileWithMethodBody(string.Format(TestInput, testInput), "Bar", out var semanticModel);
+            var methodSymbol = semanticModel.GetDeclaredSymbol(method);
+            var varDeclarators = method.DescendantNodes().OfType<VariableDeclaratorSyntax>();
+            var myTupleSymbol = semanticModel.GetDeclaredSymbol(varDeclarators.First(d => d.Identifier.ToString() == "myTuple"));
+            var cSymbol = semanticModel.GetDeclaredSymbol(varDeclarators.First(d => d.Identifier.ToString() == "c"));
+
+            var cfg = CSharpControlFlowGraph.Create(method.Body, semanticModel);
+            var lva = CSharpLiveVariableAnalysis.Analyze(cfg, methodSymbol, semanticModel);
+            var explodedGraph = new CSharpExplodedGraph(cfg, methodSymbol, semanticModel, lva);
+
+            var numberOfProcessedInstructions = 0;
+            explodedGraph.InstructionProcessed +=
+                (sender, args) =>
+                {
+                    numberOfProcessedInstructions++;
+                    if (args.Instruction.ToString() == "myTuple = (1, 2)")
+                    {
+                        args.ProgramState.GetSymbolValue(myTupleSymbol).Should().NotBeNull();
+                    }
+                    if (args.Instruction.ToString() == "c = (1, null)")
+                    {
+                        args.ProgramState.GetSymbolValue(cSymbol).Should().NotBeNull();
+                    }
+
+                    // Symbolic value for tuples are in the stack for compatibility
+                    if (args.Instruction.ToString() == "(object d, object e)")
+                    {
+                        args.ProgramState.HasValue.Should().BeTrue();
+                    }
+                    // Stack is clean after assignment
+                    if (args.Instruction.ToString() == "(object d, object e) = (1, null)")
+                    {
+                        args.ProgramState.HasValue.Should().BeFalse();
+                    }
+                };
+
+            explodedGraph.Walk();
+            numberOfProcessedInstructions.Should().Be(7);
+        }
```

Comment:
I think it would be great to have a test also for using declarations and one for statements.

Reply:
I added a test for local using statement, but in the `ControlFlowGraphTest` class, as it is not really related to the eploded graph. Other tests for usings are already there